File: Activity2.1/app/src/main/java/com/example/activirty21/MainActivity.java
package com.example.activirty21;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;
import android.os.Handler;
import android.os.Looper;
import android.os.Message;
import android.util.Log;
import android.view.View;
import android.widget.Toast;


import java.io.IOException;
import java.util.ArrayList;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

public class MainActivity extends AppCompatActivity implements View.OnClickListener, Handler.Callback {

    private RecyclerView recyclerView;
    private ArrayList<String>name, hobby, age, phone, address;
    private Handler dataHandler;

    // URL to get contacts JSON
    private static String url = "https://raw.githubusercontent.com/JuanPyV/DispositivosMoviles/master/Activity2.1/app/src/main/res/JSON/amiwos.json";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        recyclerView = findViewById(R.id.recyclerView);

        recyclerView.setHasFixedSize(true);

        dataHandler = new Handler(Looper.getMainLooper(), this);

        name = new ArrayList<>();
        hobby = new ArrayList<>();
        age = new ArrayList<>();
        phone = new ArrayList<>();
        address = new ArrayList<>();
    }




    public void llenarRecycler(View v) throws IOException, JSONException {


        Request r = new Request("https://raw.githubusercontent.com/JuanPyV/DispositivosMoviles/master/Activity2.1/app/src/main/res/JSON/amiwos.json", dataHandler);
        r.start();


        LinearLayoutManager llm = new LinearLayoutManager(this);
        llm.setOrientation(LinearLayoutManager.VERTICAL);

        GridLayoutManager glm = new GridLayoutManager(this, 2);

        recyclerView.setLayoutManager(llm);

        friendAdapter adapter =new friendAdapter(name, hobby, this);

        recyclerView.setAdapter(adapter);

    }

    @Override
    public boolean handleMessage(@NonNull Message msg) {
        try {
            JSONArray datos = (JSONArray) msg.obj;
            //Toast.makeText(this, datos.toString(), Toast.LENGTH_SHORT).show();
            for(int i = 0; i < datos.length(); i++){

                JSONObject actual = datos.getJSONObject(i);
                name.add(actual.getString("name"));
                hobby.add(actual.getString("hobby"));
                age.add(actual.getInt("age") + "");
                phone.add(actual.getString("phone"));
                address.add(actual.getString("address"));
            }

        } catch (JSONException e) {
            e.printStackTrace();
        }
        return true;

    }


    @Override
    public void onClick(View v) {

        int pos = recyclerView.getChildLayoutPosition(v);
        Toast.makeText(this, name.get(pos) + " " +
                hobby.get(pos) + " " +
                age.get(pos) + " " +
                phone.get(pos) + " " +
                address.get(pos), Toast.LENGTH_LONG).show();
    }
}

File: Kotlin-JSON/app/src/main/java/com/example/kotlinjson/MainActivity.kt
package com.example.kotlinjson

import android.os.Bundle
import android.util.Log
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import kotlin.math.log

class MainActivity : AppCompatActivity(){

    private lateinit var recyclerView:RecyclerView ;
    private val comidasDeMemin = listOf(
        Comida("burger", 15.0, "a juicy burger!"),
        Comida("hotdog", 20.0, "an ok hot dog"),
        Comida("tacos", 12.0, "some pretty good tacos"),
        Comida("torta", 22.0, "nice torta"),
        Comida("carne asada", 50.0, "a great carne asada")


    )

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_recycler)

        recyclerView = findViewById<RecyclerView>(R.id.recyclerView)
        recyclerView.setHasFixedSize(true)



        recyclerView.adapter = FoodAdapter(comidasDeMemin) { comida : Comida -> comidaItemClicked(comida) }

        val llm = LinearLayoutManager(this)
        llm.orientation = LinearLayoutManager.VERTICAL

        GridLayoutManager(this, 2)

        recyclerView.layoutManager = llm


    }

    private fun comidaItemClicked(comida : Comida) {
        Toast.makeText(this, comida.description, Toast.LENGTH_LONG).show()
    }




}
